### PR TITLE
Fix: service user avatar appeared as group conversation grid

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

After https://github.com/wireapp/wire-ios/pull/3493, a conversation with a service user only appears as a group conversation.

### Causes

All group conversations are displayed in `.four` Mode no matter the type of `users`

### Solutions

Refactor the logic, when there is only one user in a group conversation, display it in `.one` mode.
Add tests for the `Mode` enum.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift [hunk 1]
```diff
@@ -107,13 +107,24 @@ extension Mode {
     ///   - conversationType: when conversationType is nil, it is a incoming connection request
     ///   - users: number of users involved in the conversation
     fileprivate init(conversationType: ZMConversationType? = nil, users: [UserType]) {
-        switch (conversationType, users.count) {
-        case (.group?, 1...):
+        let isServiceUser: Bool
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift]: Why not move this inside the `case .group?, 1:`

like

```
self = users[0].isServiceUser ?  .one(serviceUser: true) : .four
```

File: Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift [hunk 2]
```diff
@@ -107,13 +107,24 @@ extension Mode {
...
+        if users.count > 0 {
+            isServiceUser = users[0].isServiceUser
+        } else {
+            isServiceUser = false
+        }
+
+        switch (conversationType, users.count, isServiceUser) {
+        case (.group?, 1, true),
+             (.oneOnOne?, _, _),
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift]: A service user can never be inside `oneOnOne` or `.connection` so we can remove these cases.

File: Wire-iOS Tests/ConversationAvatarViewTests.swift [hunk 3]
```diff
@@ -35,6 +35,84 @@ extension Array where Element: ZMUser {
     }
 }
 
+protocol CoreDataFixed {
```
Comment (typfel) [inline Wire-iOS Tests/ConversationAvatarViewTests.swift]: This will saves us some typing but I think we should discuss what we want to achieve with this protocol. I'd prefer if we do this in separate PR.

File: Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift [hunk 4]
```diff
@@ -171,7 +182,7 @@ final public class ConversationAvatarView: UIView {
         }
     }
     
-    private var mode: Mode = .one(serviceUser: false) {
+    var mode: Mode = .one(serviceUser: false) {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift]: Let's make the setter private?

File: Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Components/Views/ConversationAvatarView.swift]: The logic is refactored and the if case is moved inside the switch case.

File: Wire-iOS Tests/ConversationAvatarViewTests.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS Tests/ConversationAvatarViewTests.swift]: remove the test for now
